### PR TITLE
feat(table-of-contents): add self-filtering functionality

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

Related to: #10443 

### Description

This PR adds a small additional filter to the TOC's `_setTargets` method, preventing elements with the`no-toc` attribute from being added to the TOC.

### Changelog

**New**

- Table of Contents filters out elements with `no-toc` attribute

### CODE REVIEW

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 1]
```diff
@@ -376,9 +376,9 @@ class DDSTableOfContents extends HostListenerMixin(
             elem.innerText.match(notWhiteSpace) ||
             elem.dataset.title?.match(notWhiteSpace);
           const hasNameAttr = elem.matches(selectorTarget);
-          const notExcluded = !tagNamesToAvoid.includes(
-            elem.tagName.toLowerCase()
-          );
+          const notExcluded =
+            !tagNamesToAvoid.includes(elem.tagName.toLowerCase()) &&
+            !elem.hasAttribute('no-toc');
```
Comment (FlashGooden) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:381]: Ok, so we should be able to just drop this attribute in the `<form />`? 🪄

**Review by annawen1 — APPROVED** [review]
LGTM

**Review by jkaeser — COMMENTED** [review]
@andy-blum Does this block specific elements, or will it prevent the ToC from finding targets within an entire DOM subtree?

**Comment by jkaeser** [comment]
@andy-blum I wonder if it should block subtrees. Then it doesn't matter what issues adopters run into; they will have the power to disable whole chunks of the DOM. This is important when considering DOM output the adopter doesn't necessarily control (such as when using third party code). The current solution implies either:
1. The third party needs to change its output to avoid a visual error when used in conjunction with some framework outside of its control (backwards logic)
2. The adopter needs to write a script to modify third party code (potentially heavy lift)

**Comment by andy-blum** [comment]
I'm fine with that. @carbon-design-system/carbon-for-ibm-com-admins do you have any opinons?

**Pull request merged**